theta_dup/include/theta_sketch_dup.h

```diff
@@ -0,0 +1,2209 @@
+   * @param value float to be removed from the sketch
+   */
+  void remove(float value);
+
+  /**
+   * Remove one element of any type from this sketch.
+   * This is a "universal" remove that covers all cases above,
+   * but may produce different hashes.
+   * Be very careful to hash input values consistently using the same approach
+   * both over time and on different platforms
+   * and while passing sketches between C++ environment and Java environment.
+   * Otherwise two sketches that should represent overlapping sets will be
+   * disjoint For instance, for signed 32-bit values call update(int32_t) method
+   * above, which does widening conversion to int64_t, if compatibility with
+   * Java is expected
+   * @param data pointer to the data
+   * @param length of the data in bytes
+   */
+  void remove(const void* data, unsigned length);
+
+  /**
+   * Remove retained entries in excess of the nominal size k (if any)
+   */
+  void trim();
+
+  /**
+   * Converts this sketch to a compact sketch (ordered or unordered).
+   * @param ordered optional flag to specify if ordered sketch should be
+   * produced
+   * @return compact sketch
+   */
+  compact_theta_sketch_dup_alloc<A> compact(bool ordered = true) const;
+
+  virtual typename theta_sketch_dup_alloc<A>::const_iterator begin() const;
+  virtual typename theta_sketch_dup_alloc<A>::const_iterator end() const;
+
+  /**
+   * This method deserializes a sketch from a given stream.
+   * @param is input stream
+   * @param seed the seed for the hash function that was used to create the
+   * sketch
+   * @return an instance of a sketch
+   */
+  static update_theta_sketch_dup_alloc<A> deserialize(
+      std::istream& is, uint64_t seed = DEFAULT_SEED);
+
+  /**
+   * This method deserializes a sketch from a protobuf
+   * @param pb input stream
+   * @param seed the seed for the hash function that was used to create the
+   * sketch
+   * @return an instance of a sketch
+   */
+  static update_theta_sketch_dup_alloc<A> deserialize(
+      datasketches_pb::ThetaSketchDup& pb,
+      uint64_t seed = DEFAULT_SEED);
+
+  /**
+   * This method deserializes a sketch from a given array of bytes.
+   * @param bytes pointer to the array of bytes
+   * @param size the size of the array
+   * @param seed the seed for the hash function that was used to create the
+   * sketch
+   * @return an instance of the sketch
+   */
+  static update_theta_sketch_dup_alloc<A> deserialize(
+      const void* bytes, size_t size, uint64_t seed = DEFAULT_SEED);
+
+  /**
+   * @return true if *this equals r
+   */
+  virtual bool isEqual(const update_theta_sketch_dup_alloc& r) const;
+
+ private:
+  // resize threshold = 0.5 tuned for speed
+  static constexpr double RESIZE_THRESHOLD = 0.5;
+  // hash table rebuild threshold = 15/16
+  static constexpr double REBUILD_THRESHOLD = 15.0 / 16.0;
+
+  static constexpr uint8_t STRIDE_HASH_BITS = 7;
+  static constexpr uint32_t STRIDE_MASK = (1 << STRIDE_HASH_BITS) - 1;
```

Comment:
Any comments regarding stride?

Reply:
Updated. See [this commit](https://github.com/googleinterns/cardinality-estimation-with-data-deletions/pull/2/commits/3ad49957b9d057931610fce078283b3f108593e3).